src/Compilers/CSharp/Portable/Symbols/SymbolDistinguisher.cs

```diff
@@ -103,12 +103,13 @@ private void MakeDescriptions()
                     // May not be the case if there are error types.
                     if ((object)containingAssembly0 != null && (object)containingAssembly1 != null)
                     {
+                        // Use the assembly identities rather than locations. Note that the
+                        // assembly identities may be identical as well. (For instance, the
+                        // symbols are type arguments to the same generic type, and the type
+                        // arguments have the same string representation. The assembly
+                        // identities will refer to the generic types, not the type arguments.)
                         location0 = containingAssembly0.Identity.ToString();
                         location1 = containingAssembly1.Identity.ToString();
-
-                        // Even if the friendly locations produced by GetLocationString aren't
-                        // distinct, the containing assembly identities should be.
-                        Debug.Assert(location0 != location1);
```

Comment:
It feels like augmenting description with the same location doesn't make diagnostics less confusing and the added test actually demonstrates this. I would say that we should leave descriptions as is in this case.